src/typesystem.hpp

```diff
@@ -34,7 +36,7 @@ struct MonoData {
 };
 
 // A variable is just a name for a different monotype.
-// VarData stores a MonoID that indicates which monotype it is equal
+// VarData stores a MonoId that indicates which monotype it is equal
 // to. It can also indicate that it is equal to itself, meaning that
 // we don't know its concrete type.
 struct VarData {
```

Comment:
You might want to merge master into your branch before you get too many conflicts